### PR TITLE
[cmake][rocRAND] Updated toolchain-linux.cmake to have proper CMAKE_PREFIX_PATH definitions

### PR DESCRIPTION
**Pull request by NguyenNhuDi**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,11 @@ Documentation for rocRAND is available at
 * `--test|-t` is no longer a required flag for `rtest.py`. Instead, the user can use either `--emulation|-e` or `--test|-t`, but not both.
 * Removed TBB dependency for multi-core processing of host-side generation.
 
+## Fixes
```
Comment (amd-jnovotny) [inline CHANGELOG.md]:
```suggestion
## Resolved issues
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,11 @@ Documentation for rocRAND is available at
...
+
+* Fixed an issue where `CMAKE_PREFIX_PATH` was not defined properly in `CMAKELists.txt` and `toolchain-linux.cmake`.
+* Fixed an issue where `cmake_platform_opts` in `rmake.py` was sometimes a string instead of a list.
```
Comment (amd-jnovotny) [inline CHANGELOG.md]:
```suggestion
* Fixed an issue in `rmake.py` where `cmake_platform_opts` was sometimes a string instead of a list.
```

**Review by amd-jnovotny — APPROVED** [review]
Changelog looks good overall. Just one section header change and an editorial suggestion.

**Comment by NguyenNhuDi** [comment]
Updated various CMAKE files.

- Moved the position of variable definitions to the top in CMAKELists
- Edited toolchain-linux to define CMAKE_PREFIX_PATH as well
- Removed a part of Dependencies.cmake where it forces the CMAKE_CXX_COMPILER for benchmarks to be cl or g++ if it was not GNU

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,9 +14,15 @@ Documentation for rocRAND is available at
 
 ### Changed
 
+* Removed a section in `cmake/Dependencies.cmake` that was forcing `DCMAKE_CXX_COMPILER` to be either `cl` or `g++` if the compiler was not `GNU`.
```
Comment (amd-jnovotny) [inline CHANGELOG.md]:
```suggestion
* Removed a section in `cmake/Dependencies.cmake` that was forcing `DCMAKE_CXX_COMPILER` to be set to either `cl` or `g++` if the compiler was not `GNU`.
```

**Review by stanleytsang-amd — APPROVED** [review]

**Pull request merged**